docs/adr/0006-process-for-onboarding-new-tenants.md

```diff
@@ -0,0 +1,299 @@
+# Process for onboarding new tenants and enrolling them to use Marain services
+
+## Status
+
+Proposed
+
+## Context
+
+We have defined (in ADR 0006) the way in which we intend to implement Tenancy in Marain instances using the `Marain.Tenancy` service. As noted in that ADR, managing the desired model by hand would be excessively error prone and as such, we need to design tooling that will allow us to create and manage new tenants, and to allow them to use the Marain services they are licenced for.
+
+Before we can build that tooling we need to design the underlying process by which tenant onboarding, enrollment and offboarding will work. This needs to allow new Client Tenants to be onboarded into Marain services without tightly coupling the services so some central thing that knows everything about them.
+
+## Decision
+
+We are envisaging a central control-plane API (referred to for the remainder of this document as the "Management API") for Marain which primarily builds on top of the Tenancy service. This will provide the standard operations such as creating new tenants and enrolling them to use Marain services.
+
+It will also need to allow us to manage licencing, but we anticipate that this will be via an as-yet nonexistent API, as it will be a central service much like tenancy, used by all other services (either directly or via something like APIM).
```

Comment:
Is it just licensing? It appears that Licensing & Metering (for billing) are often bedfellows...

Reply:
Also - what's your definition of licensing? Are you talking about access control? Rights management? Different feature tiers or SKUs? It feels like this might need a better definition.

Reply:
Yes - billing, metering, licensing (and authorization) are three (four) different things. It is perhaps better to call out that we are not really touching on those topics here. Merely "enrollment in the service". And we have worked hard to keep them orthogonal.